### ISSUE TITLE
save fails if model's primary key is not auto-assigned and called 'id'

### ISSUE THREAD

**Issue by pryankster** [bfirsh/django-ordered-model#29]
If you have a model which has a manually-assigned primary key, it will not be 'None' going into save.  Also, if that key isn't called 'id', the OrderedModel.Save() method will fail.

It also fails setting order when there are not records in the db already, because:

self.order = c and c + 1 or 0

fails if c is 0 (like it will be with an empty table)

Here's a fix (ordered_model\models.py):

```
    def save(self, *args, **kwargs):
        # print "save: %r %r" % (self.__class__, self.pk)
        # used to be 'if not self.key', but that fails when primary key 
        # isn't called 'id'.  so tried 'if not self.pk', but that fails 
        # when pk isn't autoassigned.
        if self.order is None:
            c = self.__class__.objects.all().aggregate(Max('order')).get('order__max')
            # this doesn't catch c == None.
            # self.order = c and c + 1 or 0
            self.order = c is not None and c + 1 or 0
            # print "c = %r" % (c,)
        # print "order=%r" % ( self.order, )
        super(OrderedModel, self).save(*args, **kwargs)
```

**Comment by Zopieux** [comment]
Isn't the issue fixed by #34? Could you close if it is?

**Comment by shuckc** [comment]
We no longer assume the model has a field called 'id' anywhere in the OrderedModel code, so this should be fixed. There is one remaining place in the Admin that needs fixing. Please re-open if you have problems.
